src/dumbo/src/tcp/endpoint.rs

```diff
@@ -299,16 +303,36 @@ impl Endpoint {
     }
 }
 
+/// `parse_request` parses the request bytes and builds a `micro_http::Response` by the given callback function.
+pub fn parse_request(
+    request_bytes: &[u8],
+    callback: fn(Result<Request, RequestError>) -> Response,
+) -> Response {
+    let request = Request::try_from(request_bytes);
+    callback(request)
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
 
     use std::fmt;
     use std::str::from_utf8;
 
+    use micro_http::{Request, RequestError, Response, StatusCode, Version};
     use pdu::tcp::Flags as TcpFlags;
     use tcp::connection::tests::ConnectionTester;
 
+    fn mock_callback(request: Result<Request, RequestError>) -> Response {
+        match request {
+            Ok(request) => {
+                let _uri = request.uri().get_abs_path();
+                Response::new(Version::Http11, StatusCode::OK)
+            }
+            Err(_) => Response::new(Version::Http11, StatusCode::BadRequest),
+        }
```

Comment:
```suggestion
            // If we really need some request based processing in the mock callback, we should do it here.
           // Otherwise, return a default response.
           Response::new(Version::Http11, StatusCode::OK)
        }
```